tmt/utils.py

```diff
@@ -2252,10 +2252,15 @@ def run(command: str) -> str:
             '/', os.path.relpath(fmf_root, git_root))
 
     # Get the ref (skip for the default)
-    ref = run('git rev-parse --abbrev-ref HEAD')
     def_branch = default_branch(git_root)
-    if ref != def_branch or always_get_ref:
-        fmf_id.ref = ref
+    if def_branch is None:
+        fmf_id.ref = None
+    else:
+        ref = run('git rev-parse --abbrev-ref HEAD')
+        if ref != def_branch or always_get_ref:
+            fmf_id.ref = ref
+        else:
```

Comment:
Why do we set `ref` to the `def_branch` when `always_get_ref` is false? 

When `ref == def_branch` we should keep `fmf_id.ref = None` so consumers can do shallow git clone.

Reply:
Updated it to be `None` when `ref == def_branch`.